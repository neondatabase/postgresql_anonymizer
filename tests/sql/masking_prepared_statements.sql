BEGIN;

CREATE EXTENSION anon;

SET anon.transparent_dynamic_masking TO TRUE;

CREATE TABLE t AS SELECT n FROM generate_series(1,100) n;

SECURITY LABEL FOR anon ON COLUMN t.n
  IS 'masked with VALUE 0';

CREATE ROLE dumper LOGIN;

security label for anon on role dumper
  is 'masked';

GRANT SELECT ON TABLE t TO dumper;

PREPARE max_value AS
  SELECT n
  FROM t
  ORDER BY n DESC
  LIMIT 1
;

EXECUTE max_value;

SET ROLE dumper;

SELECT n = 0
FROM t
ORDER BY n DESC
LIMIT 1;

SAVEPOINT before_execute;
  EXECUTE max_value;
ROLLBACK TO before_execute;

DEALLOCATE PREPARE max_value;

ROLLBACK;
