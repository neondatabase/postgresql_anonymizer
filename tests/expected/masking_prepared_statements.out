BEGIN;
CREATE EXTENSION anon;
SET anon.transparent_dynamic_masking TO TRUE;
CREATE TABLE t AS SELECT n FROM generate_series(1,100) n;
SECURITY LABEL FOR anon ON COLUMN t.n
  IS 'masked with VALUE 0';
CREATE ROLE dumper LOGIN;
security label for anon on role dumper
  is 'masked';
GRANT SELECT ON TABLE t TO dumper;
PREPARE max_value AS
  SELECT n
  FROM t
  ORDER BY n DESC
  LIMIT 1
;
EXECUTE max_value;
  n  
-----
 100
(1 row)

SET ROLE dumper;
SELECT n = 0
FROM t
ORDER BY n DESC
LIMIT 1;
 ?column? 
----------
 t
(1 row)

SAVEPOINT before_execute;
  EXECUTE max_value;
ERROR:  Anon: role is masked
ROLLBACK TO before_execute;
DEALLOCATE PREPARE max_value;
ROLLBACK;
